### DISCUSSION TITLE
How to sync results on multiple devices? And API key

### DISCUSSION THREAD

**Discussion by xmexpi** [anse-app/anse#92] [General]
Could you please guide me on syncing this application so that all results are securely saved on multiple devices? Is it feasible to achieve this using an API code, similar to SQL or some other method? Currently, I have it deployed on Vercel.

**Comment by ddiu8081** [comment]
Currently, all of anse's settings and history are stored locally in the browser (specifically, indexedDB).

Setting items, dialog lists, and history messages can be exported as Json data respectively. You can find function `exportData` in `src/stores/settings.ts`, `src/stores/conversation.ts` and `src/stores/message.ts`. After exporting the Json data, you can upload and synchronize them in any way you wish.

https://github.com/anse-app/anse/blob/79a1c7262bc82a3053a75938dcc6ce48eec73cd5/src/stores/storage/conversation.ts#L30-L37

After that, you can modify the code in the app's startup stage to download this data. The function is `rebuildStores` in `src/stores/db.ts`:

https://github.com/anse-app/anse/blob/79a1c7262bc82a3053a75938dcc6ce48eec73cd5/src/stores/storage/db.ts#L18-L22
